api/external/secrets/secrets.go

```diff
@@ -77,6 +78,8 @@ func (s *Secret) Validate() error {
 	case "service_now":
 		errors = requiredField(kvMap["username"], requiredServiceNowUsernameError, errors)
 		errors = requiredField(kvMap["password"], requiredServiceNowPasswordError, errors)
+	case "chef-server":
+		errors = requiredField(kvMap["key"], requiredChefServerOrganizationKeyError, errors)
```

Comment:
- As discussed have added type `chef-server` but it subject to change. 
- At present, it's required to add validation for admin `key` only.